### PR TITLE
ModifyPlayerZoneStatistic and UpdatePlayerStatistic GM Updates

### PR DESCRIPTION
**Pull request by EmosewaMC**
These GMs are server handled GMs, not client side as seen [here](https://github.com/DarkflameUniverse/DarkflameServer/blob/09dfb6df3a53ce3fd3870351e8b3e6633e5a349b/dGame/dGameMessages/GameMessageHandler.cpp#L652-L654)

There is also a missing enum which I have defined here as well and updated the GM to use.

### CODE REVIEW

**Comment by EmosewaMC** [comment]
Also spotted a second GM related to statistics that is a server received GM as referenced in the same link above

**Comment by lcdr** [comment]
DLU code is unfortunately not authoritative on whether a packet is supposed to be handled by the server, the captures are. In any case, a packet being handled by the server does not mean it is not also handled by the client, it can be both. In the case of ModifyPlayerZoneStatistic, I can indeed find packets for both ways in the captures, but for UpdatePlayerStatistic, I can only find server->client packets, not client->server ones. However the captures aren't exhaustive, so if you can indeed get the client to send this packet it'd be reason to include it.

Packets that can be sent both ways should not be defined in either of gm/client/mod.rs or gm/server/mod.rs, but instead be defined in gm/mod.rs, and imported and included in the client and server enums. There should also be tests for both, just duplicate the existing ones.

File: src/world/gm/server/mod.rs [hunk 1]
```diff
@@ -873,6 +886,45 @@ pub struct PlayerRailArrivedNotification {
 	pub waypoint_number: i32,
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq, GmParam)]
+#[repr(i32)]
+pub enum StatisticID {
```
Comment (lcdr) [inline src/world/gm/server/mod.rs:891]: Keeping with Rust naming convention this should be `StatisticId`

**Comment by EmosewaMC** [comment]
Very interesting.  I've been used to "if we handle a packet, the response is a different one.  Perhaps this is instead a dlu oversight and this packet needs to be sent to the client whenever a statistic is updated?

**Comment by EmosewaMC** [comment]
Addressed issues with the PR
